### PR TITLE
Use threshold computation from Raptor library

### LINKED ISSUE #48: Use threshold computation from Raptor library
https://github.com/seqan/raptor/pull/148

### PR DESCRIPTION
**Pull request by eaasna**
Resolves #48 and resolves https://github.com/eaasna/valik/issues/42 which was caused by a bug that was since fixed in raptor

### CODE REVIEW

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -7,6 +7,8 @@ cmake_minimum_required (VERSION 3.8)
 add_library ("${PROJECT_NAME}_interface" INTERFACE)
 target_link_libraries ("${PROJECT_NAME}_interface" INTERFACE seqan3::seqan3)
 target_include_directories ("${PROJECT_NAME}_interface" INTERFACE ../include)
+target_include_directories ("${PROJECT_NAME}_interface" INTERFACE ../lib/raptor/src)
```
Comment (eseiler) [inline src/CMakeLists.txt:10]:
```suggestion
```

File: src/CMakeLists.txt [hunk 2]
```diff
@@ -19,17 +21,8 @@ add_library ("${PROJECT_NAME}_build_lib" STATIC valik_build.cpp)
 target_link_libraries ("${PROJECT_NAME}_build_lib" PUBLIC "${PROJECT_NAME}_interface")
 
 # Sliding window search
```
Comment (eseiler) [inline src/CMakeLists.txt:23]:
```suggestion
# Sliding window search
add_library ("raptor_threshold" STATIC
             ../lib/raptor/src/threshold/one_indirect_error_model.cpp
             ../lib/raptor/src/threshold/multiple_error_model.cpp
             ../lib/raptor/src/threshold/pascal_row.cpp
             ../lib/raptor/src/threshold/one_error_model.cpp
             ../lib/raptor/src/threshold/precompute_correction.cpp
             ../lib/raptor/src/threshold/precompute_threshold.cpp
)
target_link_libraries ("raptor_threshold" PUBLIC "${PROJECT_NAME}_interface")
```

File: src/CMakeLists.txt [hunk 3]
```diff
@@ -19,17 +21,8 @@ add_library ("${PROJECT_NAME}_build_lib" STATIC valik_build.cpp)
...
-
-add_library ("${PROJECT_NAME}_minimiser_model_lib" STATIC search/minimiser_model.cpp)
-target_link_libraries ("${PROJECT_NAME}_minimiser_model_lib" PUBLIC "${PROJECT_NAME}_search_helper_lib")
-
-add_library ("${PROJECT_NAME}_simple_model_lib" STATIC search/compute_simple_model.cpp)
-target_link_libraries ("${PROJECT_NAME}_simple_model_lib" PUBLIC "${PROJECT_NAME}_minimiser_model_lib")
-
 add_library ("${PROJECT_NAME}_search_lib" STATIC valik_search.cpp)
-target_link_libraries ("${PROJECT_NAME}_search_lib" PUBLIC "${PROJECT_NAME}_simple_model_lib")
+target_link_libraries ("${PROJECT_NAME}_search_lib" PUBLIC "${PROJECT_NAME}_interface")
```
Comment (eseiler) [inline src/CMakeLists.txt:25]:
```suggestion
target_link_libraries ("${PROJECT_NAME}_search_lib" PUBLIC "raptor_threshold")
```

**Review by eseiler — COMMENTED** [review]
This should probably work

File: test/api/valik/search/local_prefilter_test.cpp [hunk 4]
```diff
@@ -91,7 +93,7 @@ TEST(make_pattern_bounds, first_pattern_of_query)
     valik::search_arguments arguments{};
     arguments.pattern_size = 12;
     arguments.window_size = 8;
-    arguments.shape_size = 4;
+    arguments.shape = seqan3::ungapped(4);
```
Comment (marehr) [inline test/api/valik/search/local_prefilter_test.cpp:96]:
```suggestion
    arguments.shape = seqan3::ungapped{4};
```

**Review by marehr — APPROVED** [review]

**Pull request merged**